src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -1595,6 +1595,11 @@ private BoundExpression BindDefaultExpression(DefaultExpressionSyntax node, Bind
                 // Note, this call can clear and reuse lookupResult and members
                 reportPrimaryConstructorParameterShadowing(node, symbol ?? members[0], name, invoked, lookupResult, members, diagnostics);
                 members.Free();
+
+                if (symbol is not SynthesizedAccessorValueParameterSymbol { Name: "value" })
```

Comment:
>if (symbol is not SynthesizedAccessorValueParameterSymbol { Name: "value" }) [](http://example.com/codeflow?start=16&length=76)

Consider placing this check closer to where the identifier is bound, i.e. right after the ```GetSymbolOrMethodOrPropertyGroup``` call.